docs/guides/miscellaneous/how-to-write-a-how-to-guide.md

```diff
@@ -2,4 +2,214 @@
+
+- Snippets should be as short as possible, but no shorter. In general, you can think of the snippet like a diff: what needs to change to accomplish this step?
+
+- Steps should be linear. “Do A, then B, then C.” Avoid complex loops and/or branching. If loops or branching are needed, it is likely a sign that the scope of the guide is too big. In that case, consider options for splitting it into more than one how-to guide.
+
+**Additional notes**: This section covers errata that would be distracting to include in Steps. It’s fine for it to be empty.
+
+**Additional resources**: Additional resources, usually external (i.e. not within the Great Expectations documentation) and usually shown as a list. To avoid link rot, please use this section sparingly, and prefer links to stable, well-maintained resources.
+
+Once a draft of your guide is written, you can see it rendered on a `localhost` by adding your document to the table-of-contents. In Docusaurus this is done by the [sidebar.js file](https://docusaurus.io/docs/sidebar).
+
+
+## Steps
+
+
+:::warning
+  Links are broken. Fix the following
+  - [Setting up your dev environment](./how-to-template)
+  - [Style guide](./how-to-template)
+  - [Testing](./how-to-template)
+:::
+
+
+1. Ensure that your dev environment is set up according to the guide on [Setting up your dev environment](#TODO)
+
+2. Copy the How-to guide template file to the appropriate subdirectory of `docs/guides/`, and rename it.
+
+3. Write a title and purpose paragraph.
+
+4. Decide whether you’re writing a code-heavy or process-heavy guide, and adjust your formatting appropriately.
+
+5. Fill out the Prerequisites info box (see [How-to guide template file](./how-to-template)). The header of the info box says: “This how-to guide assumes you have already:”. Place each prerequisite under its own bullet and phrase it using the style in the template: “done something”.
+
+6. Fill in the Steps section, making sure to include bash, yml, and code snippets as appropriate.
+    - These will typically be included in a separate file that is in the `great_expectations/test/integrations` folder and be referenced in the how-to-doc. For additional details, please see the "Structure of a How-to-guide" section below.  
+
+7. If needed, add content to Additional Notes and/or Additional Resources. These sections supplement the article with information that would be distracting to include in Steps. It’s fine for them to be empty.
+
+8. Scan your article to make sure it follows the [Style guide](#TODO). If you’re not familiar with the Style Guide, that’s okay: your PR reviewer will also check for style and let you know if we find any issues.
+
+9. Locally run integration tests for any code that was included as part of the guide. Also see our guide on [Testing](#TODO)
+
+10. Submit your PR! If there are any additional integrations that need to be run, then please add this to your PR message.
+
+## Purpose of a how-to guide
```

Comment:
Could we put the Purpose before much earlier in this doc? In the case of the Rule-Based Profiler stuff, I think it would have helped to read this first before embarking on writing anything.